plugins/modules/uptimerobot.py

```diff
@@ -25,133 +27,132 @@
-    req, info = fetch_url(module, full_uri)
-    result = to_text(req.read())
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
+    response = requests.post(API_URL + action, headers=headers, data=payload)
 
+    if response.status_code == 200:
+        return response.json()
+    else:
+        raise Exception('API request failed',
+                        response.status_code, response.text)
 
-def pauseMonitor(module, params):
-
-    params['monitorStatus'] = 0
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = to_text(req.read())
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
-
-
-def main():
-
-    module = AnsibleModule(
-        argument_spec=dict(
-            state=dict(required=True, choices=['started', 'paused']),
-            apikey=dict(required=True, no_log=True),
-            monitorid=dict(required=True)
-        ),
-        supports_check_mode=SUPPORTS_CHECK_MODE
-    )
-
-    params = dict(
-        apiKey=module.params['apikey'],
-        monitors=module.params['monitorid'],
-        monitorID=module.params['monitorid'],
-        format=API_FORMAT,
-        noJsonCallback=API_NOJSONCALLBACK
-    )
-
-    check_result = checkID(module, params)
 
-    if check_result['stat'] != "ok":
-        module.fail_json(
-            msg="failed",
-            result=check_result['message']
+if __name__ == '__main__':
+    from ansible.module_utils.basic import AnsibleModule
+
+    def main():
+        module = AnsibleModule(
+            argument_spec=dict(
+                api_key=dict(type='str', required=True, aliases=['apikey']),
+                action=dict(
+                    type='str',
+                    required=True,
+                    choices=[
+                        'getAccountDetails',
+                        'getMonitors',
+                        'newMonitor',
+                        'editMonitor',
+                        'deleteMonitor',
+                        'resetMonitor',
+                        'getAlertContacts',
+                        'newAlertContact',
+                        'editAlertContact',
+                        'deleteAlertContact',
+                        'getMWindows',
+                        'newMWindow',
+                        'editMWindow',
+                        'deleteMWindow',
+                        'getPSPs',
+                        'newPSP',
+                        'editPSP',
+                        'deletePSP',
+                        # for backwards compatibility
+                        'started',
+                        'paused'
+                    ],
+                    aliases=['state']
+                ),
+                params=dict(type='dict', required=False),
+                # for backwards compatibility
+                state=dict(type='str', required=False),
```

Comment:
`state` is already an alias of `action`. You cannot have `state` both as a separate action and as an alias.